### PR TITLE
windows: root build

### PR DESCRIPTION
**Pull request by majcherm-da**
- All bazel targets' build & test on Windows.
- Fixes Haskell bindings tests on Windows
- Fixes `daml_test` rule to make it work on Windows
- Marks `daml_test` flaky on Windows (related to https://github.com/digital-asset/daml/issues/1907)

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -30,47 +30,12 @@ function bazel() {
...
-    //rules_daml/... `
-    //scala-protoc-plugins/... `
-    //templates/...
+bazel build `-`-experimental_execution_log_file ${ARTIFACT_DIRS}/build_execution_windows.log //...
 
 bazel shutdown
 
-bazel run `
-    //daml-foundations/daml-tools/da-hs-damlc-app `-`- `-h
+bazel run //daml-foundations/daml-tools/da-hs-damlc-app `-`- `-h
```
Comment (cocreature) [inline build.ps1]: I think we can just kill this line. We have tests that exercise this code path now, e.g, the integration tests. This was only added when we first got damlc to compile.

File: rules_daml/daml.bzl [hunk 2]
```diff
@@ -104,8 +104,16 @@ daml_compile = rule(
 
 def _daml_test_impl(ctx):
     script = """
-      {damlc} test --files {files}
-    """.format(damlc = ctx.executable.damlc.short_path, files = " ".join([f.short_path for f in ctx.files.srcs]))
+      set -eou pipefail
+
+      DAMLC=$(rlocation $TEST_WORKSPACE/{damlc})
```
Comment (aherrmann-da) [inline rules_daml/daml.bzl]: How come this doesn't require the usual [bash `rlocation` boilerplate](https://github.com/bazelbuild/bazel/blob/168ee7c589955372843115f67d9df2b4cc42e6ba/examples/shell/test.sh#L18)?

Comment (majcherm-da) [reply rules_daml/daml.bzl]: I was also a bit surprised, but it's injected automatically - the difference here is that it's a Bazel's rule, not plain bash script used in `sh_test`. Generated test exec files does not contain any defintion of the `rlocation`, so execution of these tests outside of Bazel runtime will fail, but not sure if we care about such case (running tests generated by Bazel outside of Bazel).

Comment (aherrmann-da) [reply rules_daml/daml.bzl]: Interesting, that's good to know.

> not sure if we care about such case (running tests generated by Bazel outside of Bazel).

The environment between `bazel run` and `bazel test` is typically different, so I guess it's fine if these tests require `bazel test`. It's still possible to pass arguments using `--test_args` if that's necessary at some point.

**Pull request merged**